build.gradle.kts

```diff
@@ -65,7 +65,7 @@ tasks.processResources {
 
 allprojects {
     tasks.withType<Jar>().configureEach {
-        archiveVersion.set(null as String?)
+        archiveFileName.set(provider { "${this@configureEach.archiveBaseName.get()}.jar" })
```

Comment:
Wow, this is some weird notation. Can we do this in some other way that makes it easier to understand?

Reply:
What weird notation are you referring to? `this@configureEach`?

I can assign `this` to a `val` and then use the `val` in the inner lambda.

Reply:
There are several levels of indirection to unpack here. I think separating some of it out would be a good idea. Introducing a `val` for `this` could work. Though can't you simply refer to `${archiveBaseName.get()}`?

Reply:
I simplified it to
```
archiveFileName.set(archiveBaseName.map { "$it.jar" })
```

Reply:
Looks much cleaner, thanks.